sdks/java/core/src/main/java/org/apache/beam/sdk/coders/BigIntegerCoder.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.coders;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.math.BigInteger;
+
+/**
+ * A {@link BigIntegerCoder} encodes a {@link BigInteger} as a length-prefixed
+ * byte array containing the big endian two's-complement representation.
+ */
+public class BigIntegerCoder extends AtomicCoder<BigInteger> {
+
+  @JsonCreator
+  public static BigIntegerCoder of() {
+    return INSTANCE;
+  }
+
+  /////////////////////////////////////////////////////////////////////////////
+
+  private static final BigIntegerCoder INSTANCE = new BigIntegerCoder();
+
+  private BigIntegerCoder() {}
+
+  @Override
+  public void encode(BigInteger value, OutputStream outStream, Context context)
+      throws IOException, CoderException {
+    checkNotNull(value, String.format("cannot encode a null %s", BigInteger.class.getSimpleName()));
```

Comment:
I say "double check" because as far as our project & IMO modern Java is concerned, if it isn't `@Nullable` it isn't nullable.